core/src/ldml/ldml_processor.cpp

```diff
@@ -225,77 +225,84 @@ ldml_processor::process_event(
           state->actions().push_backspace(KM_KBP_BT_UNKNOWN);
         } else {
           state->actions().push_backspace(KM_KBP_BT_CHAR, last_char);
+          state->context().pop_back();
         }
-        state->context().pop_back();
```

Comment:
if the context is empty (see prior loop/comment), pop_back() understandably throws